push_notifications/apns_async.py

```diff
@@ -1,14 +1,17 @@
 import asyncio
 from dataclasses import asdict, dataclass
 import time
-from typing import Dict, Union
+from typing import Awaitable, Callable, Dict, Optional, Union
 
-from aioapns import APNs, NotificationRequest, ConnectionError
+from aioapns import APNs, NotificationRequest, ConnectionError, NotificationResult
 
 from . import models
 from .conf import get_manager
 from .exceptions import APNSServerError
 
+ErrFunc = Optional[Callable[[NotificationRequest, NotificationResult], Awaitable[None]]]
```

Comment:
Out of office right now, `Awaitable` is that some form of `Future` result?

Reply:
#Update: Interesting, 

https://docs.python.org/3/library/typing.html#annotating-callable-objects

I have always take a different approach to this!